enumerable.rb

```diff
@@ -0,0 +1,151 @@
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = flatten if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    for i in arr
+      yield(i)
+    end
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return self unless block_given?
+
+    arr = check_data_type(self)
+
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  ## my_select method
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return true if yield(i) }
+    false
+  end
+
+  ## my_all method
+  def my_all
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_none method
+  def my_none
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |i| return false unless yield(i) }
+    true
+  end
+
+  ## my_count method
+  def my_count
+    return self unless block_given?
+
+    arr = self
+    counter = 0
+    arr.my_each { |i| counter += 1 if yield(i) }
+    counter
+  end
+
+  ## my_map method
+  def my_map(&factor)
+    return self unless block_given?
+
+    arr = self
+    new_arr = []
+    arr.my_each { |x| new_arr << factor.call(x) }
+    new_arr
+  end
+
+  ## my_inject method
+  def my_inject(&factor)
+    return self unless block_given?
+
+    arr = self
+    arr.my_each { |x| factor.call(x) }
+  end
+
+  ## multiply_els method
+  def multiply_els
+    return self unless block_given?
+
+    arr = self
+    result = 1
+    arr.my_inject { |i| result = yield(result, i) }
+    result
+  end
```

Comment:
- [ ] enumerables #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject
- [ ] Since #multiply_els is not an enumerable method, move the function out of the module enumerable